assertj-tests/assertj-integration-tests/assertj-scripts/src/test/java/org/assertj/scripts/ShellScriptInvoker.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2025 the original author or authors.
+ */
+package org.assertj.scripts;
+
+import org.junit.jupiter.api.io.TempDir;
+import org.junit.platform.commons.support.Resource;
+
+import static java.lang.Runtime.getRuntime;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.nio.file.Files.copy;
+import static java.nio.file.Files.readString;
+import static java.nio.file.Files.writeString;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.util.Arrays.array;
+import static org.junit.platform.commons.support.ReflectionSupport.findAllResourcesInPackage;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+/**
+ * A class that invokes the shell script and compare the output.
+ *
+ * <p>It will use a temporary file to write the input and call the script, then it will compare the file to the expected value</p>
+ * @author XiaoMingZHM, Eveneko
+ */
+abstract class ShellScriptInvoker {
+
+  @TempDir
+  private Path tempDir;
```

Comment:
This is quite inefficient for the scope of such tests – I'll refactor it to make sure we create one temporary directory per class.